### ISSUE TITLE
[JENKINS-53462] Jenkins websites use non-trusted 'submit' event to start form submission when current browser is Firefox

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#8648]
<p>HTML spec [<span class="error">&#91;1&#93;</span>|<a href="https://w3c.github.io/uievents/#trusted-events" class="external-link" target="_blank" rel="nofollow noopener">https://w3c.github.io/uievents/#trusted-events</a>] says "Most untrusted events will not trigger default actions, with the exception of the click event.". Now Firefox doesn't comply with the spec. When I try to fix the bug [<span class="error">&#91;2&#93;</span>|<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630" class="external-link" target="_blank" rel="nofollow noopener">https://bugzilla.mozilla.org/show_bug.cgi?id=1370630</a>], a regression has happened on all Jenkins websites. Users can't login Jenkins websites with Firefox anymore. After some experiments, it seems the Jenkins websites detect the browser's user agent and use untrusted 'submit' event to start form submission when the current browser is Firefox. Changing the UA of Chrome to the same string as Firefox also block the form submission.</p>

<p> </p>

<p>The steps I used to reproduce this problem</p>

<p>On Chrome</p>
<ol>
	<li>Change UA to the same string as Firefox</li>
	<li>Navigate <a href="https://jenkins.qa.ubuntu.com/" class="external-link" target="_blank" rel="nofollow noopener">https://jenkins.qa.ubuntu.com/</a></li>
	<li>Click login</li>
	<li>Enter username/password and press 'log in' button</li>
	<li>Nothing happened</li>
</ol>

<p>Expectation</p>

<p>Don't use untrusted events to start form submission on Jenkins websites.</p>

<p> </p>

<p><span class="error">&#91;1&#93;</span> <a href="https://w3c.github.io/uievents/#trusted-events" class="external-link" target="_blank" rel="nofollow noopener">https://w3c.github.io/uievents/#trusted-events</a></p>

<p><span class="error">&#91;2&#93;</span> <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630" class="external-link" target="_blank" rel="nofollow noopener">https://bugzilla.mozilla.org/show_bug.cgi?id=1370630</a></p>

<p> </p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=iamstone">iamstone</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-53462" target="_blank">Jenkins websites use non-trusted 'submit' event to start form submission when current browser is Firefox</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>label(s)</b>: 2.164.3-fixed
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2021-04-20T13:29:16+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 9
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>HTML spec [<span class="error">&#91;1&#93;</span>|<a href="https://w3c.github.io/uievents/#trusted-events" class="external-link" target="_blank" rel="nofollow noopener">https://w3c.github.io/uievents/#trusted-events</a>] says "Most untrusted events will not trigger default actions, with the exception of the click event.". Now Firefox doesn't comply with the spec. When I try to fix the bug [<span class="error">&#91;2&#93;</span>|<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630" class="external-link" target="_blank" rel="nofollow noopener">https://bugzilla.mozilla.org/show_bug.cgi?id=1370630</a>], a regression has happened on all Jenkins websites. Users can't login Jenkins websites with Firefox anymore. After some experiments, it seems the Jenkins websites detect the browser's user agent and use untrusted 'submit' event to start form submission when the current browser is Firefox. Changing the UA of Chrome to the same string as Firefox also block the form submission.</p>

<p> </p>

<p>The steps I used to reproduce this problem</p>

<p>On Chrome</p>
<ol>
	<li>Change UA to the same string as Firefox</li>
	<li>Navigate <a href="https://jenkins.qa.ubuntu.com/" class="external-link" target="_blank" rel="nofollow noopener">https://jenkins.qa.ubuntu.com/</a></li>
	<li>Click login</li>
	<li>Enter username/password and press 'log in' button</li>
	<li>Nothing happened</li>
</ol>

<p>Expectation</p>

<p>Don't use untrusted events to start form submission on Jenkins websites.</p>

<p> </p>

<p><span class="error">&#91;1&#93;</span> <a href="https://w3c.github.io/uievents/#trusted-events" class="external-link" target="_blank" rel="nofollow noopener">https://w3c.github.io/uievents/#trusted-events</a></p>

<p><span class="error">&#91;2&#93;</span> <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630" class="external-link" target="_blank" rel="nofollow noopener">https://bugzilla.mozilla.org/show_bug.cgi?id=1370630</a></p>

<p> </p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>classic login form (before 2.128), regular &quot;Save&quot; form submission buttons on the classic UI</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348673&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348673">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I cannot, and could never, despite Firefox being my default browser, reproduce the issue on any Jenkins instance, and it is unclear that the issue as reported actually exists in current releases of Jenkins. More information is needed before we even consider this to be an open bug.</p>
</pre>
</details>

</details>
<p>I cannot, and could never, despite Firefox being my default browser, reproduce the issue on any Jenkins instance, and it is unclear that the issue as reported actually exists in current releases of Jenkins. More information is needed before we even consider this to be an open bug.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348679&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348679">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>OK. The bug currently in Firefox allows the login to happen, which is why I never encountered the described problem. Once the bug in Firefox is fixed, it will no longer be able to use the login form. Now I got it.</p>

<p>Would be interesting to know whether simply giving Firefox the 'Chrome behavior' (if any), i.e. modifying or removing browser detection, would make this work. That should be fairly straightforward, I hope.</p>

<p>I'll move this to the correct Jira project.</p>
</pre>
</details>

</details>
<p>OK. The bug currently in Firefox allows the login to happen, which is why I never encountered the described problem. Once the bug in Firefox is fixed, it will no longer be able to use the login form. Now I got it.</p>

<p>Would be interesting to know whether simply giving Firefox the 'Chrome behavior' (if any), i.e. modifying or removing browser detection, would make this work. That should be fairly straightforward, I hope.</p>

<p>I'll move this to the correct Jira project.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348680&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348680">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> Can you clarify what release of Firefox this should be in? From my reading, it should be in 56.x, but I'm on 61, soon 62, and can log in just fine, current or previous login form.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> Can you clarify what release of Firefox this should be in? From my reading, it should be in 56.x, but I'm on 61, soon 62, and can log in just fine, current or previous login form.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348685&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348685">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>We had to backout the patch due to the regression with Jenkins. As such it was only available in some Nightly builds of Firefox during the 56 development process.</p>
</pre>
</details>

</details>
<p>We had to backout the patch due to the regression with Jenkins. As such it was only available in some Nightly builds of Firefox during the 56 development process.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348687&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348687">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Ok, so I checked with such a Nightly build and switched the user agent to Chrome. That actually fixed it, yes.</p>
</pre>
</details>

</details>
<p>Ok, so I checked with such a Nightly build and switched the user agent to Chrome. That actually fixed it, yes.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348688&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348688">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I wonder what happens with Jenkins in current releases of Firefox if we remove the special treatment. I assume that was added for a reason.</p>

<p>IOW, do we rip out the Firefox special behavior, or do we need more elaborate Firefox special behavior to remain compatible with current and future Firefox.</p>
</pre>
</details>

</details>
<p>I wonder what happens with Jenkins in current releases of Firefox if we remove the special treatment. I assume that was added for a reason.</p>

<p>IOW, do we rip out the Firefox special behavior, or do we need more elaborate Firefox special behavior to remain compatible with current and future Firefox.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/top12345tw.svg" title="top12345tw's avatar" /> top12345tw's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348693&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348693">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I checked the LTS 2.121.2 and 2.121.3, I still saw the Firefox special behavior. I am not sure if 2.138.1 fixes this, but I am happy to test again.</p>

<p>From the form submission point of view, it is not necessary to give Firefox special behavior, giving the same behavior as other browser should just also work on Firefox.</p>
</pre>
</details>

</details>
<p>I checked the LTS 2.121.2 and 2.121.3, I still saw the Firefox special behavior. I am not sure if 2.138.1 fixes this, but I am happy to test again.</p>

<p>From the form submission point of view, it is not necessary to give Firefox special behavior, giving the same behavior as other browser should just also work on Firefox.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348695&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348695">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Should be testable by setting a different User-Agent on a prod release of Firefox I suppose.</p>
</pre>
</details>

</details>
<p>Should be testable by setting a different User-Agent on a prod release of Firefox I suppose.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348696&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348696">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>And assuming Chrome behaves to spec (like Firefox post-fix), we should be able to observe the problem in Jenkins by setting Chrome's User-Agent to Firefox.</p>
</pre>
</details>

</details>
<p>And assuming Chrome behaves to spec (like Firefox post-fix), we should be able to observe the problem in Jenkins by setting Chrome's User-Agent to Firefox.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/top12345tw.svg" title="top12345tw's avatar" /> top12345tw's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348698&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348698">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Or you can test this by setting User-Agent to Firefox on Chrome.</p>
</pre>
</details>

</details>
<p>Or you can test this by setting User-Agent to Firefox on Chrome.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=348699&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348699">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<blockquote><p>And assuming Chrome behaves to spec (like Firefox post-fix), we should be able to observe the problem in Jenkins by setting Chrome's User-Agent to Firefox.</p></blockquote>
<p>This is exactly what's happening.</p>

<p>The good news is that the redesigned login form will work in 2.138.1.</p>

<p>The bad news is that all other submit buttons will not. It's not quite as bad as before, as most forms have an 'Apply' button that still works, but close.</p>
</pre>
</details>

</details>
<blockquote><p>And assuming Chrome behaves to spec (like Firefox post-fix), we should be able to observe the problem in Jenkins by setting Chrome's User-Agent to Firefox.</p></blockquote>
<p>This is exactly what's happening.</p>

<p>The good news is that the redesigned login form will work in 2.138.1.</p>

<p>The bad news is that all other submit buttons will not. It's not quite as bad as before, as most forms have an 'Apply' button that still works, but close.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=349167&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349167">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Regarding the bad news, about how many different submit buttons we are talking here? Is it something which could easily be fixed and shipped, or how long would it take? So far Jenkins seems to be the only site which blocks us from shipping this feature fix, and we really would like to get this out to our users for a better web platform conformance.</p>

<p> </p>

<p>Thanks.</p>
</pre>
</details>

</details>
<p>Regarding the bad news, about how many different submit buttons we are talking here? Is it something which could easily be fixed and shipped, or how long would it take? So far Jenkins seems to be the only site which blocks us from shipping this feature fix, and we really would like to get this out to our users for a better web platform conformance.</p>

<p> </p>

<p>Thanks.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=349227&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349227">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I suspect it'll be one button, but it's reused a lot as the <tt>f:submit</tt> form control. Just needs someone to figure out where JS takes a wrong turn to address this.</p>
</pre>
</details>

</details>
<p>I suspect it'll be one button, but it's reused a lot as the <tt>f:submit</tt> form control. Just needs someone to figure out where JS takes a wrong turn to address this.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=351019&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-351019">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I will have a look now whether I can come up with a fix.</p>

<p>My understanding so far:</p>
<ul class="alternate" type="square">
	<li>I need a patched FF version (the earlier links do not work anymore, can you <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> provide me with a link?)</li>
	<li>The "new login" page works, as <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> said, most likely because I only used vanilla js, no external libs and especially no 
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">f:submit</pre>
</div></div></li>
	<li>in the bugzilla thread someone mentioned YUI I think that may be the base problem.</li>
</ul>


<p>I will now review the </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">f:submit</pre>
</div></div>
<p> and see whether I can extract all usage of YUI and do "vanilla js" submits. </p>
</pre>
</details>

</details>
<p>I will have a look now whether I can come up with a fix.</p>

<p>My understanding so far:</p>
<ul class="alternate" type="square">
	<li>I need a patched FF version (the earlier links do not work anymore, can you <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> provide me with a link?)</li>
	<li>The "new login" page works, as <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> said, most likely because I only used vanilla js, no external libs and especially no 

<pre>
f:submit</pre></li>
	<li>in the bugzilla thread someone mentioned YUI I think that may be the base problem.</li>
</ul>


<p>I will now review the </p>

<pre>
f:submit</pre>
<p> and see whether I can extract all usage of YUI and do "vanilla js" submits. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=351025&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-351025">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a></p>
<blockquote><p>I need a patched FF version</p></blockquote>
<p>Not quite. All you need is a Chrome that claims to be FF, then that will encounter the same problem as patched/future Firefox would. Probably simpler to get started <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a></p>
<blockquote><p>I need a patched FF version</p></blockquote>
<p>Not quite. All you need is a Chrome that claims to be FF, then that will encounter the same problem as patched/future Firefox would. Probably simpler to get started <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=351029&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-351029">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://github.com/jenkinsci/jenkins/pull/3689" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/3689</a> the hack version</p>
</pre>
</details>

</details>
<p><a href="https://github.com/jenkinsci/jenkins/pull/3689" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/3689</a> the hack version</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=351064&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-351064">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Jenkins PR build for download at <a href="https://ci.jenkins.io/job/Core/job/jenkins/job/PR-3689/2/artifact/org/jenkins-ci/main/jenkins-war/2.147-rc27437.1590be3ff083/jenkins-war-2.147-rc27437.1590be3ff083.war" class="external-link" target="_blank" rel="nofollow noopener">https://ci.jenkins.io/job/Core/job/jenkins/job/PR-3689/2/artifact/org/jenkins-ci/main/jenkins-war/2.147-rc27437.1590be3ff083/jenkins-war-2.147-rc27437.1590be3ff083.war</a></p>

<p>Perhaps those watchers with ability to patch and build their own Firefox could check whether form submission buttons in this Jenkins build address the problem?</p>

<p> </p>
</pre>
</details>

</details>
<p>Jenkins PR build for download at <a href="https://ci.jenkins.io/job/Core/job/jenkins/job/PR-3689/2/artifact/org/jenkins-ci/main/jenkins-war/2.147-rc27437.1590be3ff083/jenkins-war-2.147-rc27437.1590be3ff083.war" class="external-link" target="_blank" rel="nofollow noopener">https://ci.jenkins.io/job/Core/job/jenkins/job/PR-3689/2/artifact/org/jenkins-ci/main/jenkins-war/2.147-rc27437.1590be3ff083/jenkins-war-2.147-rc27437.1590be3ff083.war</a></p>

<p>Perhaps those watchers with ability to patch and build their own Firefox could check whether form submission buttons in this Jenkins build address the problem?</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=351603&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-351603">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a>, I assume that without your patch saving the settings doesn't work? Also AFAIR it was mentioned that all submit buttons throughout Jenkins are affected by that. They all work that way now? If yes, that sounds great.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a>, I assume that without your patch saving the settings doesn't work? Also AFAIR it was mentioned that all submit buttons throughout Jenkins are affected by that. They all work that way now? If yes, that sounds great.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=351666&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-351666">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>As final explanation why FF stopped working is that since we did not provide a callback to the onClick, FF correctly stops propagating and all other browser are wrong to propagate the event to a submit.</p>

<p>We fixed the f:submit however the problem may occur in other parts but the solution will always be to provide a default fallback for onClick.</p>
</pre>
</details>

</details>
<p>As final explanation why FF stopped working is that since we did not provide a callback to the onClick, FF correctly stops propagating and all other browser are wrong to propagate the event to a submit.</p>

<p>We fixed the f:submit however the problem may occur in other parts but the solution will always be to provide a default fallback for onClick.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=351815&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-351815">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>That's great to hear that you are considering it for a LTS release even it's not the upcoming one. Thanks a lot!</p>
</pre>
</details>

</details>
<p>That's great to hear that you are considering it for a LTS release even it's not the upcoming one. Thanks a lot!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=352772&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-352772">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a>, I simply want to report back that one of our engineers who works a lot with Jenkins just checked the latest release of Jenkins and it indeed seems to work all fine. Having it in the next LTS release would be fantastic! Thanks!</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a>, I simply want to report back that one of our engineers who works a lot with Jenkins just checked the latest release of Jenkins and it indeed seems to work all fine. Having it in the next LTS release would be fantastic! Thanks!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=353247&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353247">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>What I wonder is that the change here should have only affected Firefox, right? But people are filing issues for Chrome and IE.</p>
</pre>
</details>

</details>
<p>What I wonder is that the change here should have only affected Firefox, right? But people are filing issues for Chrome and IE.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=353794&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353794">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Proposed reversal of the fix in <a href="https://github.com/jenkinsci/jenkins/pull/3760" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/3760</a> due to regressions it caused (especially for the upcoming 2.150.x LTS line).</p>

<p>Proposed amendment of the fix in <a href="https://github.com/jenkinsci/jenkins/pull/3761" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/3761</a> in the hopes it would solve the problem. Due to time constraints this has undergone minimal testing, would appreciate if others could take a look.
<a href="https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/main/jenkins-war/2.152-rc27510.acc4eec688a0/" class="external-link" target="_blank" rel="nofollow noopener">https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/main/jenkins-war/2.152-rc27510.acc4eec688a0/</a> has the jenkins.war for this PR, which currently is 2.151 + the specific change.</p>
</pre>
</details>

</details>
<p>Proposed reversal of the fix in <a href="https://github.com/jenkinsci/jenkins/pull/3760" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/3760</a> due to regressions it caused (especially for the upcoming 2.150.x LTS line).</p>

<p>Proposed amendment of the fix in <a href="https://github.com/jenkinsci/jenkins/pull/3761" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/3761</a> in the hopes it would solve the problem. Due to time constraints this has undergone minimal testing, would appreciate if others could take a look.<br/>
<a href="https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/main/jenkins-war/2.152-rc27510.acc4eec688a0/" class="external-link" target="_blank" rel="nofollow noopener">https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/main/jenkins-war/2.152-rc27510.acc4eec688a0/</a> has the jenkins.war for this PR, which currently is 2.151 + the specific change.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=364824&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-364824">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>2.164.3, scheduled for May 8, the earliest, if accepted as a backport. The next .1, scheduled for early June, otherwise. Assuming of course we don't cause regressions again.</p>
</pre>
</details>

</details>
<p>2.164.3, scheduled for May 8, the earliest, if accepted as a backport. The next .1, scheduled for early June, otherwise. Assuming of course we don't cause regressions again.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olivergondza.png" title="olivergondza's avatar" /> olivergondza's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=365299&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-365299">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Can someone verify the patch? I have failed to reproduce the problem using <tt>firefox-nightly-68.0a1.20190422-1</tt> and <tt>jenkins/jenkins:2.172</tt>. Both logging in as admin and saving global config page works for me.</p>
</pre>
</details>

</details>
<p>Can someone verify the patch? I have failed to reproduce the problem using <tt>firefox-nightly-68.0a1.20190422-1</tt> and <tt>jenkins/jenkins:2.172</tt>. Both logging in as admin and saving global config page works for me.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=365301&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-365301">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza" class="user-hover" rel="olivergondza">olivergondza</a>, a normal nightly build of Firefox won't work given that it hasn't the patch from <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630" class="external-link" target="_blank" rel="nofollow noopener">bug 1370630</a> included. You can pick a build from <a href="https://treeherder.mozilla.org/#/jobs?repo=try&amp;revision=3f12e17b558a36ef5b65235d663c4e617c3d328e&amp;searchStr=build" class="external-link" target="_blank" rel="nofollow noopener">https://treeherder.mozilla.org/#/jobs?repo=try&amp;revision=3f12e17b558a36ef5b65235d663c4e617c3d328e&amp;searchStr=build</a> by just clicking the `B` symbol of your platform, selecting Job Details in the lower pane, and then `target.zip`, `target.dmg`, or `target.tar.bz2`. Just extract the archive and run the binary. With such a build you should be able to see the failure without the patch in Jenkins.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza" class="user-hover" rel="olivergondza">olivergondza</a>, a normal nightly build of Firefox won't work given that it hasn't the patch from <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630" class="external-link" target="_blank" rel="nofollow noopener">bug 1370630</a> included. You can pick a build from <a href="https://treeherder.mozilla.org/#/jobs?repo=try&revision=3f12e17b558a36ef5b65235d663c4e617c3d328e&searchStr=build" class="external-link" target="_blank" rel="nofollow noopener">https://treeherder.mozilla.org/#/jobs?repo=try&revision=3f12e17b558a36ef5b65235d663c4e617c3d328e&searchStr=build</a> by just clicking the `B` symbol of your platform, selecting Job Details in the lower pane, and then `target.zip`, `target.dmg`, or `target.tar.bz2`. Just extract the archive and run the binary. With such a build you should be able to see the failure without the patch in Jenkins.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=365305&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-365305">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza" class="user-hover" rel="olivergondza">olivergondza</a> AFAIUI, the behavior of Chrome is the correct one (i.e. future Firefox's), so you should be able to confirm using Chrome with a Firefox user agent string. That's how I reproduced it and worked on the fix.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza" class="user-hover" rel="olivergondza">olivergondza</a> AFAIUI, the behavior of Chrome is the correct one (i.e. future Firefox's), so you should be able to confirm using Chrome with a Firefox user agent string. That's how I reproduced it and worked on the fix.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=365306&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-365306">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> Thanks for the link. "Save" buttons don't work on the nightly build and Jenkins 2.164.2, as expected. And they work on Jenkins 2.174, also as expected <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/biggrin.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> Thanks for the link. "Save" buttons don't work on the nightly build and Jenkins 2.164.2, as expected. And they work on Jenkins 2.174, also as expected <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/biggrin.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=365312&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-365312">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Great to be helpful. And that sounds fantastic. So we now have to wait for the 2.174 and an upcoming LTS before we can make that change live in Firefox.</p>
</pre>
</details>

</details>
<p>Great to be helpful. And that sounds fantastic. So we now have to wait for the 2.174 and an upcoming LTS before we can make that change live in Firefox.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olivergondza.png" title="olivergondza's avatar" /> olivergondza's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=365315&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-365315">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Cool, thanks! I am backporting this to 2.164.3 to be released <del>2019-04-08</del> (EDIT: 2019-05-08), unless further regressions are found.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a>, I do appreciate the effort put in making sure Jenkins is ready for the fix in Firefox. Though I am wondering if there is a chance to postpone the delivery until significant Jenkins (or what other applications are affected) userbase migrate to the fixed version.</p>
</pre>
</details>

</details>
<p>Cool, thanks! I am backporting this to 2.164.3 to be released <del>2019-04-08</del> (EDIT: 2019-05-08), unless further regressions are found.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a>, I do appreciate the effort put in making sure Jenkins is ready for the fix in Firefox. Though I am wondering if there is a chance to postpone the delivery until significant Jenkins (or what other applications are affected) userbase migrate to the fixed version.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=365324&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-365324">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> We are unfamiliar with the Firefox release schedule, could you clarify what that means in terms of rolling out the change to regular users? Those who are on 66.0.3 right now?</p>

<p>For Jenkins, based on usage statistics, 60% of LTS users are within the last 4 LTS releases (~4 months), and 50% of weekly release users are within the last 16 releases (~3.5 months). So I expect the majority of Jenkins users to be on compatible releases around August/September, and they <em>can</em> update no later than May (for LTS), and should already be updated (for weekly releases).</p>

<p>From a project POV, we only consider the latest LTS and weekly releases (e.g. only those get security fixes) to be supported. Of course users may lag behind in updating, and for some setups, updating Jenkins is a big project. The question the Firefox team will need to answer is, how aggressively do they want to annoy Firefox+Jenkins users that update one but not the other.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> We are unfamiliar with the Firefox release schedule, could you clarify what that means in terms of rolling out the change to regular users? Those who are on 66.0.3 right now?</p>

<p>For Jenkins, based on usage statistics, 60% of LTS users are within the last 4 LTS releases (~4 months), and 50% of weekly release users are within the last 16 releases (~3.5 months). So I expect the majority of Jenkins users to be on compatible releases around August/September, and they <em>can</em> update no later than May (for LTS), and should already be updated (for weekly releases).</p>

<p>From a project POV, we only consider the latest LTS and weekly releases (e.g. only those get security fixes) to be supported. Of course users may lag behind in updating, and for some setups, updating Jenkins is a big project. The question the Firefox team will need to answer is, how aggressively do they want to annoy Firefox+Jenkins users that update one but not the other.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=365330&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-365330">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> thank you for bringing this up. It's actually not my job to make such a decision, so I will leave it up to the release managers to decide. I will clearly reference your last comment, so that they are aware of the situation for Jenkins users.</p>

<p> </p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> thank you for bringing this up. It's actually not my job to make such a decision, so I will leave it up to the release managers to decide. I will clearly reference your last comment, so that they are aware of the situation for Jenkins users.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=365541&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-365541">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Ok, so here an update. If that patch sticks and will not be backed out again, we will land the Firefox patch for Firefox 69 on May 14th or 15th. Which means it will ride the trains through beta, and will finally be released on Sep 3rd. If by that time there are still users who running a Jenkins LTS release earlier than 2.164.3, they will have to use Firefox 68 ESR. That ESR version will be maintained for another year, and will never have this patch included. Details see <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630#c41" class="external-link" target="_blank" rel="nofollow noopener">https://bugzilla.mozilla.org/show_bug.cgi?id=1370630#c41</a> and following.</p>

<p> </p>

<p>Thanks a lot again for the patch! It will unblock us from landing several features which are all blocked by this particular issue.</p>
</pre>
</details>

</details>
<p>Ok, so here an update. If that patch sticks and will not be backed out again, we will land the Firefox patch for Firefox 69 on May 14th or 15th. Which means it will ride the trains through beta, and will finally be released on Sep 3rd. If by that time there are still users who running a Jenkins LTS release earlier than 2.164.3, they will have to use Firefox 68 ESR. That ESR version will be maintained for another year, and will never have this patch included. Details see <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630#c41" class="external-link" target="_blank" rel="nofollow noopener">https://bugzilla.mozilla.org/show_bug.cgi?id=1370630#c41</a> and following.</p>

<p> </p>

<p>Thanks a lot again for the patch! It will unblock us from landing several features which are all blocked by this particular issue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=379063&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-379063">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Reopened because 2.202 backed out the change again due to the regression it caused (<a href="https://issues.jenkins.io/browse/JENKINS-58296" title="Projects are triggered twice when using File Parameter and Firefox 67" class="issue-link" data-issue-key="JENKINS-58296"><del>JENKINS-58296</del></a>).</p>
</pre>
</details>

</details>
<p>Reopened because 2.202 backed out the change again due to the regression it caused (<a href="https://issues.jenkins.io/browse/JENKINS-58296" title="Projects are triggered twice when using File Parameter and Firefox 67" class="issue-link" data-issue-key="JENKINS-58296"><del>JENKINS-58296</del></a>).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo's <a href="https://issues.jenkins.io/browse/JENKINS-53462?focusedId=401042&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-401042">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>You are right, sorry. I will try to get a clarification from our DOM folks. I will report back when there are news.</p>
</pre>
</details>

</details>
<p>You are right, sorry. I will try to get a clarification from our DOM folks. I will report back when there are news.</p>
